src/app/clusters/closure-dimension-server/closure-dimension-cluster-logic.cpp

```diff
@@ -0,0 +1,310 @@
+/**
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+/**
+ * @file Cross-platform API to handle cluster-specific logic for the valve configuration and control cluster on a single endpoint.
+ */
+
+ #include "closure-dimension-cluster-logic.h"
+
+ #include <chrono>
+ 
+ #include <app-common/zap-generated/ids/Attributes.h>
+ #include <lib/core/CHIPError.h>
+ #include <platform/CHIPDeviceLayer.h>
+ #include <system/SystemClock.h>
+ 
+ namespace chip {
+ namespace app {
+ namespace Clusters {
+ namespace ClosureDimension {
+ 
+//  void ClusterStateAttributes::Init(ClusterInitParameters state)
+//  {
+//  }
+ 
+ CHIP_ERROR ClusterStateAttributes::SetCurrent(GenericCurrentStruct & current)
+ {
+     bool dirty          = !(current == mState.current);
+     mState.current = current;
+     if (dirty)
+     {
```

Comment:
```suggestion
     if (!(mState.current == current))
     {
         mState.current = current;
```
This could be applied to other Set functions. 
Since it usesed multiple times, consider overloading the != operator in the same way you do hear, just for readability reason

Reply:
done